### PR TITLE
[RNMobile] Add optional custom editor component to native test helper

### PR DESCRIPTION
**Pull request by fluiddot**
## Description

Include an optional argument to `initializeEditor` native helper, this will allow us to provide a different Editor component when initializing the editor, which is required for tests done in Gutenberg Mobile, where we provide a custom Editor component.

## How has this been tested?

This change shouldn't affect current tests, but we could double-check that all tests pass by running the command `npm run native test`.

## Screenshots 
N/A

## Types of changes

Improvement

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

File: test/native/helpers.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ provideToNativeHtml.mockImplementation( ( html ) => {
 	serializedHtml = html;
 } );
 
-export function initializeEditor( props ) {
+export function initializeEditor( props, customEditorComponent ) {
```
Comment (dcalhoun) [inline test/native/helpers.js]: Personally, I dislike JavaScript's lack of named arguments. I find function signatures with more than one argument often become vague at the call site. I.e. `initializeEditor( ~props=testProps, ~customEditorComponent=MockEditor)` would be more explicit and an ideal interface. The closest thing in JavaScript is the pattern of an "options" or "configuration" argument. WDYT about moving `customEditorCompnent` to an options argument? 

```js
export function initializeEditor(props, { component = Editor } = {}) {
  const EditorComponent = component;
  const screen = render(<EditorComponent />);
  // ...
}
```

That would make the call site more explicit.

```js
const screen = initializeEditor(myProps, { component: MockEditor });
```

Lastly, to make this frequently used testing function even more clear, WDYT about documenting it with [JSDoc comments](https://jsdoc.app)?

```js
/**
 * Initialize an editor for test assertions.
 *
 * @param {Object}                    props               Properties passed to the editor component.
 * @param {string}                    props.initialHtml   String of block editor HTML to parse and render.
 * @param {Object}                    [options]           Configuration options for the editor.
 * @param {import('react').ReactNode} [options.component] A specific editor component to render.
 * @return {import('@testing-library/react-native').RenderAPI} A Testing Library screen.
 */
export function initializeEditor( props, { component = Editor } = {} ) {
```

**Review by dcalhoun — APPROVED** [review]
🚀 LGTM. I left one comment regarding a possible refactor to consider, but it should not be consider a blocker if you would prefer to merge as-is.

File: test/native/helpers.js (hunk 1 above)
Comment (fluiddot) [reply test/native/helpers.js]: > Personally, I dislike JavaScript's lack of named arguments. I find function signatures with more than one argument often become vague at the call site. I.e. `initializeEditor( ~props=testProps, ~customEditorComponent=MockEditor)` would be more explicit and an ideal interface. The closest thing in JavaScript is the pattern of an "options" or "configuration" argument. WDYT about moving `customEditorCompnent` to an options argument?
> 
> ```js
> export function initializeEditor(props, { component = Editor } = {}) {
>   const EditorComponent = component;
>   const screen = render(<EditorComponent />);
>   // ...
> }
> ```
> 
> That would make the call site more explicit.
> 
> ```js
> const screen = initializeEditor(myProps, { component: MockEditor });
> ```

I totally agree, to be honest, I also do dislike un-named arguments as make it harder to read and understand calls in source code 👍 . I haven't found a pattern in this regard in the code so far, nevertheless, I'd advocate following the pattern you suggested. Being this said, I'll make a quick update with your suggestion, thanks!



> astly, to make this frequently used testing function even more clear, WDYT about documenting it with [JSDoc comments](https://jsdoc.app)?
> 
> ```js
> /**
>  * Initialize an editor for test assertions.
>  *
>  * @param {Object}                    props               Properties passed to the editor component.
>  * @param {string}                    props.initialHtml   String of block editor HTML to parse and render.
>  * @param {Object}                    [options]           Configuration options for the editor.
>  * @param {import('react').ReactNode} [options.component] A specific editor component to render.
>  * @return {import('@testing-library/react-native').RenderAPI} A Testing Library screen.
>  */
> export function initializeEditor( props, { component = Editor } = {} ) {
> ```

Great, thank you so much for sharing the JSDoc for the function ❤️ , I'll add it in the next commit.

Comment (fluiddot) [reply test/native/helpers.js]: I added your suggestions in https://github.com/WordPress/gutenberg/pull/38133/commits/607b834dc03a449948c042e42c69f41299799bc3 and https://github.com/WordPress/gutenberg/pull/38133/commits/d2427dfda3eae06754ac1edeee4ed1e87d608411.

**Review by dcalhoun — APPROVED** [review]

**Pull request merged**